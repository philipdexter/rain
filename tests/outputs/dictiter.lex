keyword 'var'
name 'main'
symbol '='
keyword 'func'
symbol '('
symbol ')'
indent
keyword 'var'
name 'obj'
symbol '='
symbol '{'
name 'one'
symbol '='
string 'one'
symbol ','
name 'two'
symbol '='
string 'two'
symbol ','
symbol '['
int 3
symbol ']'
symbol '='
string 'three'
symbol ','
symbol '['
string 'fo_ur'
symbol ']'
symbol '='
string 'four'
symbol '}'
newline
keyword 'for'
name 'a'
keyword 'in'
name 'obj'
symbol ':'
name 'keys'
symbol '('
symbol ')'
indent
name 'print'
symbol '('
string 'a: '
operator '$'
name 'tostr'
symbol '('
name 'a'
symbol ')'
symbol ')'
newline
keyword 'for'
name 'b'
keyword 'in'
name 'obj'
symbol ':'
name 'keys'
symbol '('
symbol ')'
indent
name 'print'
symbol '('
string '  b: '
operator '$'
name 'tostr'
symbol '('
name 'b'
symbol ')'
symbol ')'
newline
dedent
newline
dedent
newline
name 'print'
symbol '('
string '----------------------------------------'
symbol ')'
newline
keyword 'for'
name 'a'
keyword 'in'
name 'obj'
symbol ':'
name 'values'
symbol '('
symbol ')'
indent
name 'print'
symbol '('
string 'a: '
operator '$'
name 'a'
symbol ')'
newline
keyword 'for'
name 'b'
keyword 'in'
name 'obj'
symbol ':'
name 'values'
symbol '('
symbol ')'
indent
name 'print'
symbol '('
string '  b: '
operator '$'
name 'b'
symbol ')'
newline
dedent
newline
dedent
newline
dedent
newline
EOF
