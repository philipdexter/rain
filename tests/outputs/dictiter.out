a: fo_ur
  b: fo_ur
  b: 3
  b: one
  b: two
a: 3
  b: fo_ur
  b: 3
  b: one
  b: two
a: one
  b: fo_ur
  b: 3
  b: one
  b: two
a: two
  b: fo_ur
  b: 3
  b: one
  b: two
----------------------------------------
a: four
  b: four
  b: three
  b: one
  b: two
a: three
  b: four
  b: three
  b: one
  b: two
a: one
  b: four
  b: three
  b: one
  b: two
a: two
  b: four
  b: three
  b: one
  b: two
